File: app/src/main/java/volovyk/guerrillamail/data/model/Email.java
package volovyk.guerrillamail.data.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.SerializedName;

@Entity
public class Email {
    @SerializedName("mail_from")
    private final String from;
    @SerializedName("mail_subject")
    private final String subject;
    @SerializedName("mail_body")
    private String body;
    @SerializedName("mail_date")
    private final String date;
    @PrimaryKey
    @SerializedName("mail_id")
    private final Integer id;

    public Email(String from, String subject, String body, String date, Integer id) {
        this.from = from;
        this.subject = subject;
        this.body = body;
        this.date = date;
        this.id = id;
    }

    public String getFrom() {
        return from;
    }

    public String getSubject() {
        return subject;
    }

    public String getBody() {
        return body;
    }

    public String getDate() {
        return date;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public Integer getId() {
        return id;
    }
}

File: app/src/main/java/volovyk/guerrillamail/data/model/Email.kt
package volovyk.guerrillamail.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.fasterxml.jackson.annotation.JsonProperty

@Entity
data class Email(
    @JsonProperty("mail_id") @PrimaryKey val id: String,
    @JsonProperty("mail_from") val from: String,
    @JsonProperty("mail_subject") val subject: String,
    @JsonProperty("mail_body") val body: String,
    @JsonProperty("mail_date") val date: String,
    val viewed: Boolean = false
)